components/teams-service/storage/postgres/migration/sql/07_add_db_id_field.up.sql

```diff
@@ -0,0 +1,29 @@
+BEGIN;
+
+-- break reference constraints
+ALTER TABLE teams_users_associations ADD COLUMN team_db_id SERIAL;
+ALTER TABLE teams_users_associations DROP CONSTRAINT teams_users_associations_pkey;
+ALTER TABLE teams_users_associations DROP CONSTRAINT teams_users_associations_team_id_fkey;
+
+-- massage teams for new pkey
+ALTER TABLE teams DROP CONSTRAINT teams_pkey;
+ALTER TABLE teams ADD CONSTRAINT teams_id_unique UNIQUE (id);
+ALTER TABLE teams ADD COLUMN db_id SERIAL PRIMARY KEY;
+
+-- set the team_db_id fields in the assoc table
+UPDATE teams_users_associations SET team_db_id=db_id FROM teams WHERE teams.id=team_id;
+ALTER TABLE teams_users_associations DROP COLUMN team_id;
+
+-- add new constraints
+ALTER TABLE teams_users_associations ADD CONSTRAINT teams_db_id_fkey
+  FOREIGN KEY (team_db_id) REFERENCES teams(db_id) ON DELETE CASCADE;
+ALTER TABLE teams_users_associations ADD CONSTRAINT teams_users_pkey
+  PRIMARY KEY (team_db_id, user_id);
+
+-- add helper for team db_id lookup
+CREATE FUNCTION team_db_id(_id UUID) RETURNS INTEGER AS
+$$
+  SELECT db_id FROM teams WHERE id=_id;
+$$ LANGUAGE SQL;
+
+COMMIT;
```

Comment:
I wonder if any of this is prohibitively slow with a large number of teams... 🤔

Reply:
So, to get an answer here, I've modified the `scripts/a2_collections` `authGen` helper to also add team members.

With 1000 teams having 100 members each, we find that it takes this long for running that migration:
```
[66][default:/src:0]# chef-automate dev psql chef_teams_service -- -S
psql (9.6.11)
SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
Type "help" for help.

chef_teams_service^# select count(*) from teams_users_associations ;
 count
-------
 99003
(1 row)

chef_teams_service^# select count(*) from teams;
 count
-------
  1003
(1 row)

chef_teams_service^# \q
[67][default:/src:0]# time chef-automate dev psql chef_teams_service -- -f components/teams-service/storage/postgres/migration/sql/07_add_db_id_field.up.sql
BEGIN
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
UPDATE 99003
ALTER TABLE
ALTER TABLE
ALTER TABLE
CREATE FUNCTION
COMMIT

real    0m1.035s
user    0m0.020s
sys     0m0.004s
[68][default:/src:0]#
```

Sooo... 🤷‍♂ fine.